astropy/units/equivalencies.py

```diff
@@ -561,6 +564,344 @@ def mass_energy():
                         ], "mass_energy")
 
 
+@dataclass(frozen=True)
+class GeometricUnitConversionFactors:
+    """
+    Geometric conversion factors w.r.t. SI. Implemented from
+    `Wikipedia <https://en.wikipedia.org/wiki/Geometrized_unit_system>`_.
+    """
+    m = {
+        'm': Quantity(1.0),
+        'kg': _si.c**2 / _si.G,
+        's': 1.0 / _si.c,
+        'C': _si.c**2 / (_si.G / _si.eps0)**0.5,
+        'K': _si.c**4 / (_si.G * _si.k_B)
+    }
+    kg = {
+        'm': _si.G / _si.c**2,
+        'kg': Quantity(1.0),
+        's': _si.G / _si.c**3,
+        'C': (_si.G * _si.eps0)**0.5,
+        'K': _si.c**2 / _si.k_B
+    }
+    s = {
+        'm': _si.c,
+        'kg': _si.c**3 / _si.G,
+        's': Quantity(1.0),
+        'C': _si.c**3 / (_si.G / _si.eps0)**0.5,
+        'K': _si.c**5 / (_si.G * _si.k_B)
+    }
+    C = {
+        'm': (_si.G / _si.eps0)**0.5 / _si.c**2,
+        'kg': 1.0 / (_si.G * _si.eps0)**0.5,
+        's': (_si.G / _si.eps0)**0.5 / _si.c**3,
+        'C': Quantity(1.0),
+        'K': _si.c**2 / (_si.k_B * (_si.G * _si.eps0)**0.5)
+    }
+    K = {
+        'm': _si.G * _si.k_B / _si.c**4,
+        'kg': _si.k_B / _si.c**2,
+        's': _si.G * _si.k_B / _si.c**5,
+        'C': _si.k_B * (_si.G * _si.eps0)**0.5 / _si.c**2,
+        'K': Quantity(1.0)
+    }
+
+    def _base_conversion(self, source_base, target_base):
+        """Base conversions
+
+        Parameters
+        ----------
+        source_base: astropy.unit.Unit
+            source base
+        target_base: astropy.unit.Unit
+            target base
+        """
+        try:
+            conversion_factor = getattr(self, source_base.__str__())
+        except (AttributeError, UnitTypeError):
+            raise UnitTypeError(
+                """
+                Base of source units must be one of "m", "kg", "s",
+                "C", "K".
+                """
+            )
+        return conversion_factor.get(target_base.__str__())
+
+    def _irreducible_conversion(self, source_quantity, target_unit):
```

Comment:
These functions duplicate what the unit machinery already does - you just need to define the equivalency with the units scaled the right way -- see my earlier comment.

Reply:
> there should be no need for the lambda functions.

Can you help me understand this better. As I mentioned before, there are several places, where this is done: https://github.com/astropy/astropy/pull/12847#discussion_r814135246

the third and fourth arguments of the tuple need to be callables?

Reply:
The third and fourth arguments can be omitted, in which case effectively it is treated as a pass-through function. So, conceptually, for `q.to(new_unit, equivalencies=[(unit1, unit2)])`, the quantity will first be converted to `unit1`, then its unit is replaced by `unit2`, and finally it is converted to `new_unit`. Maybe easiest to look again at https://github.com/astropy/astropy/pull/12847#discussion_r807117988 and play around with the example there in an interactive session.